packages/flutter/lib/src/cupertino/form_section.dart

```diff
@@ -0,0 +1,232 @@
+  /// order to retain the iOS look.
+  ///
+  /// The [margin] parameter sets the padding for the [Column] encapsulating
+  /// [children], and defaults to zero padding.
+  ///
+  /// The [borderRadius] parameter sets the circular border radius for the
+  /// [Column] encapsulating [children] rows. Defaults to 0.0 for the standard
+  /// edge-to-edge style.
+  ///
+  /// The [decoration] parameter sets the decoration for the section. Defaults
+  /// to a [CupertinoColors.secondarySystemBackground] background color.
+  const CupertinoFormSection({
+    Key? key,
+    required this.children,
+    this.header,
+    this.margins = EdgeInsets.zero,
+    this.borderRadius = 0.0,
+    this.decoration =
+        const BoxDecoration(color: CupertinoColors.secondarySystemBackground),
+  })  : _type = _CupertinoFormSectionType.base,
+        assert(children.length > 0),
+        super(key: key);
+
+  /// Creates a section that mimicks standard iOS forms.
+  ///
+  /// The [CupertinoFormSection.padded] constructor creates a round-edged and
+  /// padded section that is commonly seen in notched-displays like iPhone X and
+  /// beyond. Creates an iOS-style header, footer, rows, and the dividers
+  /// between rows. Does not create borders on top and bottom of the rows.
+  ///
+  /// The [header] parameter sets the form section header. The section header
+  /// lies above the [children] rows.
+  ///
+  /// The [children] parameter is required and sets the list of rows shown in
+  /// the section. The [children] parameter takes a list, as opposed to a more
+  /// efficient builder function that lazy builds, because forms are intended to
+  /// be short in row count. It is recommended that only [CupertinoFormRow] and
+  /// [CupertinoTextFormFieldRow] widgets be included in the [children] list in
+  /// order to retain the iOS look.
+  ///
+  /// The [margins] parameter sets the padding for the [Column] encapsulating
+  /// [children], and defaults to the standard notched-style iOS form padding.
+  ///
+  /// The [borderRadius] parameter sets the circular border radius for the
+  /// [Column] encapsulating [children] rows. Defaults to 10.0 for the standard
+  /// edge-to-edge style.
+  ///
+  /// The [decoration] parameter sets the decoration for the section. Defaults
+  /// to a [CupertinoColors.secondarySystemBackground] background color.
+  const CupertinoFormSection.padded({
+    Key? key,
+    required this.children,
+    this.header,
+    this.margins = _kDefaultPaddedRowsPadding,
+    this.borderRadius = 10.0,
+    this.decoration =
+        const BoxDecoration(color: CupertinoColors.secondarySystemBackground),
+  })  : _type = _CupertinoFormSectionType.padded,
+        assert(children.length > 0),
+        super(key: key);
+
+  final _CupertinoFormSectionType _type;
+
+  /// Sets the form section header. The section header lies above the
+  /// [children] rows.
+  final Widget? header;
```

Comment:
For "Plain" style `UITableView` the section header snaps to the top as the table view scrolls. To achieve that in flutter the headers need to be sliver widgets instead of regular widgets (see https://api.flutter.dev/flutter/widgets/ScrollView-class.html). Unfortunately it does not work if you put a sliver widget in `header` here, they have to be direct children of the scroll view, something like:
```dart
CustomScrollView(
  slivers: <Widget>[
    header1,
    SliverToBoxAdapter(child: CupertinoFormSection(...)),
    footer1,
    header2,
    SliverToBoxAdapter(child: CupertinoFormSection(...)),
    footer2,
  ],
) 
```
I think It would be more consistent to just leave `header` or `footer` out in this widget, and let the developer decide how these components should be put together.

Reply:
There's a difference between the traditional table view header and the settings-like forms. If you go to settings -> general -> about, you will see some headers that do not snap to the frame. This is normal behavior for forms I think. cc @xster

Reply:
I think this is ok. The normal uitableview (which this is not) is indeed more complicated.

Reply:
Oh didn't know SwiftUI had additional UI components. Taking a look.